tests/datasource/fluent/test_contexts.py

```diff
@@ -514,5 +514,30 @@ def test_data_connectors_are_built_on_config_load(
     assert dc_datasources
 
 
+@pytest.fixture
+def valid_file_path(csv_path: pathlib.Path) -> pathlib.Path:
+    return csv_path / "yellow_tripdata_sample_2018-03.csv"
+
+
+@pytest.mark.cloud
+def test_run_checkpoint_minimizes_suite_request_count(
+    seeded_cloud_context: CloudDataContext,
+    cloud_api_fake_db: FakeDBTypedDict,
+    cloud_api_fake: RequestsMock,
+    mocker: MockerFixture,
+    valid_file_path,
+):
+    validator = seeded_cloud_context.sources.pandas_default.read_csv(valid_file_path)
+    validator.expect_column_values_to_not_be_null("pickup_datetime")
+    validator.save_expectation_suite()
+    checkpoint = seeded_cloud_context.add_or_update_checkpoint(
+        name="my_quickstart_chekpoint",
+        validator=validator,
+    )
+
+    checkpoint.run()
+    # TODO assert GET expectation-suites called
```

Comment:
Did you mean to do this?

Reply:
This is turning out to be much more of a pain to setup than I expected.
Let's do this as a followup.